### PR TITLE
adding nodepool for cronjobs sbox

### PR DESCRIPTION
**Pull request by Tyler-35**
### Jira link (if applicable)
[DTSPO-15286](https://tools.hmcts.net/jira/browse/DTSPO-15286)

### Change description ###
adding additional node pool for cronjobs on the sbox env to test.

### CODE REVIEW

File: components/aks/aks.tf [hunk 1]
```diff
@@ -82,6 +82,17 @@ module "kubernetes" {
       node_taints         = []
       enable_auto_scaling = true
       mode                = "User"
+    },
+    {
+      name                = "cronjob"
+      vm_size             = lookup(var.cronjob, "vm_size", "Standard_DS3_v2")
+      min_count           = lookup(var.cronjob, "min_nodes", 0)
+      max_count           = lookup(var.cronjob, "max_nodes", 4)
```
Comment (timja) [inline components/aks/aks.tf:90]: i expect this is enough to cover all envs
```suggestion
      max_count           = lookup(var.cronjob, "max_nodes", 10)
```

File: components/aks/aks.tf [hunk 2]
```diff
@@ -82,6 +82,17 @@ module "kubernetes" {
       node_taints         = []
       enable_auto_scaling = true
       mode                = "User"
+    },
+    {
+      name                = "cronjob"
+      vm_size             = lookup(var.cronjob, "vm_size", "Standard_DS3_v2")
```
Comment (timja) [inline components/aks/aks.tf:88]: set a sensible default
```suggestion
      vm_size             = lookup(var.cronjob, "vm_size", "Standard_D4ds_v5")
```

File: environments/aks/sbox.tfvars [hunk 3]
```diff
@@ -22,6 +22,11 @@ linux_node_pool = {
   min_nodes = 4,
   max_nodes = 10
 }
+cronjob = {
+  vm_size   = "Standard_D4ds_v5",
+  min_nodes = 0,
```
Comment (timja) [inline environments/aks/sbox.tfvars:27]: this is the default so no need to specify it

File: environments/aks/sbox.tfvars [hunk 4]
```diff
@@ -22,6 +22,11 @@ linux_node_pool = {
   min_nodes = 4,
   max_nodes = 10
 }
+cronjob = {
```
Comment (timja) [inline environments/aks/sbox.tfvars:25]: I think you would need to specify this as a variable for this to work?
https://github.com/hmcts/aks-cft-deploy/blob/949fd583a23606e721625cd7492ebff0dac4d8d7/components/aks/inputs-default.tf#L17

File: environments/aks/sbox.tfvars [hunk 5]
```diff
@@ -22,6 +22,11 @@ linux_node_pool = {
...
+  vm_size   = "Standard_D4ds_v5",
+  min_nodes = 0,
+  max_nodes = 10
```
Comment (timja) [inline environments/aks/sbox.tfvars:28]: I suggested changing the default to 10 so this is unneeded

**Review by timja — COMMENTED** [review]
I don't think there's any point adding a way to override the values between envs it's unlikely to change, see my feedback below

**Review by timja — APPROVED** [review]

File: components/aks/aks.tf [hunk 6]
```diff
@@ -82,6 +82,17 @@ module "kubernetes" {
...
       mode                = "User"
+    },
+    {
+      name                = "cronjob"
+      vm_size             = "Standard_D4ds_v5"
+      min_count           = 0
+      max_count           = 10
+      max_pods            = 30
+      os_type             = "Linux"
+      node_taints         = ["kubernetes.io/name=cronjob:NoSchedule"]
```
Comment (timja) [inline components/aks/aks.tf:93]: is that name going to be right?

Comment (Tyler-35) [reply components/aks/aks.tf:93]: I still need at the toleration but plan to use that name and value.

**Review by timja — CHANGES_REQUESTED** [review]
node taint looks wrong

File: components/aks/aks.tf (hunk 6 above)
Comment (timja) [reply components/aks/aks.tf:93]: I think this may be better:
```suggestion
      node_taints         = ["dedicated=jobs:NoSchedule"]
```

Comment (Tyler-35) [reply components/aks/aks.tf:93]: didn't realise you don't need the kubernetes.io.  I'll update it now.

**Review by timja — APPROVED** [review]

**Pull request merged**